tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -908,6 +971,539 @@ static void test_rx_secure_beacon_interval(void)
+	bt_mesh_device_setup(&prov, &tx_comp);
+	test_priv_tx_init();
+
+	/* shift beaconing time line to avoid boundary cases. */
+	k_sleep(K_SECONDS(1));
+
+	PASS();
+}
+
+static void test_rx_priv_adv(void)
+{
+	struct bt_mesh_priv_beacon val;
+	int err, i;
+
+	val.enabled = 1;
+	val.rand_interval = 1;
+
+	common_rx_priv_setup();
+	bt_mesh_test_cfg_set(NULL, 200);
+
+	/* shift beaconing time line to avoid boundary cases. */
+	k_sleep(K_SECONDS(1));
+
+	err = bt_mesh_priv_beacon_cli_set(0, TX_ADDR, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
+
+	/* Because expected BEACON_INTERVAL is 10s we expect Random Value to be changed on
+	 * every beacon
+	 */
+	for (i = 0; i < 6; i++) {
+		wait_for_next_priv_beacon(BEACON_INTERVAL);
+
+		ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
+
+		memcpy(last_random, curr_random, 13);
+	}
+
+	val.rand_interval = 0;
+	err = bt_mesh_priv_beacon_cli_set(0, TX_ADDR, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
+
+	/* Because expected BEACON_INTERVAL is 10s, this should behave the same as for
+	 * Random Interval = 1
+	 */
+	for (i = 0; i < 6; i++) {
+		wait_for_next_priv_beacon(BEACON_INTERVAL);
+
+		ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
+
+		memcpy(last_random, curr_random, 13);
+	}
+
+	val.rand_interval = 3;
+	err = bt_mesh_priv_beacon_cli_set(0, TX_ADDR, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
+
+	beaconing_time = 0;
+
+	for (i = 0; i < 2; i++) {
+		for (int j = 0; j < 2; j++) {
+			wait_for_next_priv_beacon(BEACON_INTERVAL);
+
+			/* Beacon random should be the same here */
+			ASSERT_TRUE(memcmp(last_random, curr_random, 13) == 0);
+		}
+
+		wait_for_next_priv_beacon(BEACON_INTERVAL);
+
+		/* Beacon random should change here */
+		ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
+		memcpy(last_random, curr_random, 13);
+
+	}
+
+	PASS();
+}
+
+static void private_beacon_create(struct net_buf_simple *buf, const uint8_t *net_key, uint8_t flags,
+			  uint32_t iv_index)
+{
+	uint8_t beacon_key[16];
+	uint8_t net_id[8];
+	uint8_t auth[8];
+	uint8_t data[5];
+	uint8_t random_val[13];
+	int err;
+
+	err = bt_mesh_k3(net_key, net_id);
+	if (err) {
+		FAIL("Unable to generate Net ID");
+	}
+
+	err = bt_mesh_private_beacon_key(net_key, beacon_key);
+	if (err) {
+		FAIL("Unable to generate beacon key");
+	}
+
+	err = bt_mesh_beacon_auth(beacon_key, flags, net_id, iv_index, auth);
+	if (err) {
+		FAIL("Unable to generate auth value");
+	}
+
+	bt_rand(random_val, sizeof(random_val));
+	bt_mesh_beacon_encrypt(beacon_key, flags, bt_mesh.iv_index + 1,
+			       random_val, data, auth);
+
+	net_buf_simple_reset(buf);
+	net_buf_simple_add_u8(buf, BEACON_TYPE_PRIVATE);
+	net_buf_simple_add_mem(buf, random_val, 13);
+	net_buf_simple_add_mem(buf, data, 5);
+	net_buf_simple_add_mem(buf, auth, 8);
+
+}
+
+static void test_tx_priv_invalid(void)
```

Comment:
My suggestion for this test to create a test vector with offsets that need to be "corrupted" and then run the procedure through each entry of the test vector. The test vector contains offsets for both, SNB and PRB. Then we don't need to duplicate this test.

The `_rx_` side part should be identical to both beacons types as the algorithm for them is the same.

Reply:
I extracted common part as you suggested, into function that takes array of offsets and buffer to corrupt. Because TX part needs to generate different beacon type and different beacons are expected from RX side, I'm keeping these tests separate.